ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -244,6 +244,36 @@ def empty(spec: "BehaviorSpec") -> "TerminalSteps":
         )
 
 
+class ActionTuple:
+    """
+    An object whose fields correspond to actions of different types.
+    Continuous and discrete actions are numpy arrays of type float32 and
+    int32, respectively and are type checked on construction.
+    Dimensions are of (n_agents, continuous_size) and (n_agents, discrete_size),
+    respectively.
+    """
+
+    def __init__(
+        self,
+        continuous: Optional[np.ndarray] = None,
```

Comment:
I think I would prefer having this not be Optional. When the action is discrete only, the continuous action becomes of shape (n_agents, 0). I think it is slightly more general than having a None if there is no action because this way, we can eventually consider all actions to be hybrid, only some of them happen to have 0 discrete or continuous actions. @chriselion what do you think? I might be missing something

Reply:
I think for initializer arguments, keeping them Optional is nice. If you want to ensure that the fields themselves are non-None, you could "allocate" the np array on construction if they weren't provided (but not sure that's meaningful since there would be 0 elements).

As a side note, I think we're assuming that `continuous.shape[0] == discrete.shape[0]` - might be good to check that too.

Reply:
We do check that shape dimensions in `_validate_action` on the ActionSpec. It gets called in set_actions/set_actions_for_agent in the LLAPI

Reply:
`_validate_actions` will check this`